test/array.asyncMap.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const metasync = require('..');
+const metatests = require('metatests');
+
+metatests.test('succesfull map', test => {
+  test.plan(1);
+
+  const arr = [1, 2, 3];
+  const expectedArr = [2, 4, 6];
+
+  metasync.asyncMap(arr, item => item * 2, newArr => {
+    test.strictSame(newArr, expectedArr);
+  });
+});
+
+metatests.test('Non-blocking', test => {
+  test.plan(10);
+
+  const ARRAY_SIZE = 1000;
+  const arr = new Array(ARRAY_SIZE).fill(1);
+
+  const timer = setInterval(() => test.pass(), 1);
+
+  const doSmthLong = () => {
+    const begin = Date.now();
+    while (Date.now() - begin < 1);
```

Comment:
This will count 1 millisecond, was that your intention?

Reply:
Yes, is that a problem?

Reply:
No, I was just making sure.
It kind of contradicts the name, that's why I've noted it (1 ms is not 'long').